lib/livebook/hubs/enterprise_client.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Livebook.Hubs.EnterpriseClient do
+  @moduledoc false
+
+  alias Livebook.Utils.HTTP
+
+  @path "api/v1"
+
+  def fetch_info(%{"url" => url, "token" => access_token}) do
+    query = """
+    query {
+      info {
+        id
+        expire_at
+      }
+    }
+    """
+
+    with {:ok, %{"info" => info}} <- graphql(url, access_token, query) do
+      {:ok, info}
+    end
+  end
+
+  def fetch_me(%{"url" => url, "token" => access_token}) do
+    query = """
+    query {
+      me {
+        id
+      }
+    }
+    """
+
+    with {:ok, %{"me" => me}} <- graphql(url, access_token, query) do
+      {:ok, me}
+    end
+  end
+
+  defp graphql(url, access_token, query, input \\ %{}) do
+    headers = [{"Authorization", "Bearer #{access_token}"}]
+    body = {"application/json", Jason.encode!(%{query: query, variables: input})}
+
+    case HTTP.request(:post, graphql_endpoint(url), headers: headers, body: body) do
+      {:ok, 200, _, body} ->
+        case Jason.decode!(body) do
+          %{"errors" => [%{"message" => message}]} ->
+            {:error, message}
+
+          %{"data" => data} ->
+            {:ok, data}
+        end
+
+      {:ok, _, _, body} ->
+        {:error, body}
+
+      {:error, _} = error ->
+        error
```

Comment:
Unless there's a good reason not to, I would unify into `{:error, message :: String.t()}` right here. This way the caller doesn't need to match on the error term, which is coupled to how we make the request.

Reply:
I was thinking about making it specific for Enterprise coz we could say the URL is unreachable or the Token is invalid

Reply:
We can match on these here right?

Also, in case the message depends on where we call it, we could also make it a tuple like `{:error, message, type}`, where type is a standardized atom like `:not_found`, `:invalid_token`, `:other`, etc.